lua/themepark/plugins/bbox.lua

```diff
@@ -0,0 +1,282 @@
+            local sql = utils.build_select_query(columns, from, conditions, tiles.order_by, tiles.order_dir)
+
+            data.query = {{
+                sql = sql,
+            }}
+            if tiles.minzoom then
+                data.query[1].minzoom = tiles.minzoom
+            end
+            if tiles.maxzoom then
+                data.query[1].maxzoom = tiles.maxzoom
+            end
+        else
+            local conditions = assemble_conditions(minzoom, maxzoom, tiles.xycondition)
+            local sql = utils.build_select_query(columns, from, conditions, tiles.order_by, tiles.order_dir)
+            data.query = {{
+                sql = sql,
+            }}
+        end
+
+        if plugin.themepark.layer_groups[info.name] then
+            for _, group_table in ipairs(plugin.themepark.layer_groups[info.name]) do
+                local sublayer = utils.find_name_in_array(plugin.themepark.layers, group_table.name)
+                minzoom, maxzoom, columns = min_max_zoom_columns(sublayer.columns)
+
+                local conditions = assemble_conditions(minzoom, maxzoom, sublayer.tiles.xycondition)
+                local subfrom = schema_prefix .. group_table.name
+                local sql = utils.build_select_query(columns, subfrom,
+                                                     conditions, sublayer.tiles.order_by,
+                                                     sublayer.tiles.order_dir)
+                table.insert(data.query, {
+                    sql = sql,
+                    minzoom = group_table.minzoom,
+                    maxzoom = group_table.maxzoom,
+                })
+            end
+        end
+    end
+
+    return data
+end
+
+-- ---------------------------------------------------------------------------
+-- ---------------------------------------------------------------------------
+local function build_tileset_config(extra_layers)
+    local layers = {}
+
+    for _, layer in ipairs(plugin.themepark.layers) do
+        if layer.tiles ~= false then
+            local layer_config = build_layer_config(layer)
+            if layer_config then
+                table.insert(layers, layer_config)
+            end
+        end
+    end
+    for _, data in ipairs(extra_layers) do
+        if not data.geometry_field then
+            data.geometry_field = 'geom'
+        end
+        if not data.srid then
+            data.srid = plugin.themepark.options.srid
+        end
+        table.insert(layers, data)
+    end
+    return layers
+end
+
+local function ordered_keys(o)
+    local keys1 = {}
+    local keys2 = {}
+    for k, v in pairs(o) do
+        if type(v) == 'table' then
+            table.insert(keys2, k)
+        else
+            table.insert(keys1, k)
+        end
+    end
+    table.sort(keys1)
+    table.sort(keys2)
+    for _,v in ipairs(keys2) do
+        table.insert(keys1, v)
+    end
+    return keys1
+end
+
+local function dump_toml(o, indent_size, parents)
```

Comment:
We probably want to pull this out later and re-use it in the t-rex config plugin also.